### PR TITLE
Highlight JSON and SQL in Charts Editor

### PR DESCRIPTION
**Pull request by djbarnwal**
Highlight JSON and SQL in charts editor.
Using `SQLite` dialect for now for the SQL. We will have to replace this with our Metrics SQL dialect later.

image

### CODE REVIEW

File: web-common/src/components/editor/presets/yamlWithJsonAndSql.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    blockLiteralCount = 0;
+  }
+  if (!node || !activateOnNodes.has(node.name)) {
+    return null;
+  }
+  blockLiteralCount++;
+
+  if (blockLiteralCount === 1) {
+    return {
+      parser: SQLite.language.parser,
```
Comment (ericpgreen2) [inline web-common/src/components/editor/presets/yamlWithJsonAndSql.ts:22]: We've defined a CodeMirror SQL dialect for DuckDB in `web-common/src/features/models/workspace/Editor.svelte`. You could move it to a common location then import it here?

File: web-common/src/components/editor/presets/yamlWithJsonAndSql.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { jsonLanguage } from "@codemirror/lang-json";
+import { SQLite } from "@codemirror/lang-sql";
+import { yamlLanguage } from "@codemirror/lang-yaml";
+import { LRLanguage } from "@codemirror/language";
+import { parseMixed } from "@lezer/common";
+
+const activateOnNodes = new Set(["BlockLiteralContent"]);
```
Comment (ericpgreen2) [inline web-common/src/components/editor/presets/yamlWithJsonAndSql.ts:7]: Nit: a comment to explain where this "BlockLiteralContent" terminology is coming from would be helpful. Link to docs?

**Review by ericpgreen2 — APPROVED** [review]
This is great!

File: web-common/src/components/editor/presets/yamlWithJsonAndSql.ts (hunk 2 above)
Comment (djbarnwal) [reply web-common/src/components/editor/presets/yamlWithJsonAndSql.ts:7]: There is no documentation around the new YAML parser. It was added couple of months ago. This mostly came through a hit and trial using the new parser.

**Pull request merged**